gateway/src/management_switch/location_map.rs

```diff
@@ -224,9 +281,37 @@ impl ValidatedLocationConfig {
         // collection of reasons the config is invalid (if any)
         let mut reasons = Vec::new();
 
-        let names = vec_to_hashset(config.names, &mut reasons, || {
-            String::from("location names contain at least one duplicate entry")
-        });
+        let description = {
+            let n = config.description.len();
+            let description =
+                config.description.into_iter().collect::<IdOrdMap<_>>();
+
+            // Validate that we have no duplicate location names.
+            if n != description.len() {
+                reasons.push(String::from(
+                    "location descriptions contain at least one duplicate name",
```

Comment:
should we perhaps include the duplicate name here? i suppose that could be confusing if it's just the first one?